pkg/cloudprovider/aws/constraints.go

```diff
@@ -64,3 +68,12 @@ func (c *Constraints) GetLaunchTemplate() *LaunchTemplate {
 		Version: &version,
 	}
 }
+
+func (c *Constraints) GetArchitecture() string {
+	for aws, kube := range AWSToKubeArchitectures {
+		if *c.Architecture == kube {
+			return aws
+		}
+	}
+	return ""
```

Comment:
We're using this map in both directions, so this code has to exist in one form or another. I actually inverted it from what it used to be (minor cleanup), since it allowed us to remove a bit of code.